linode/lke/resource.go

```diff
@@ -96,7 +96,18 @@ func readResource(ctx context.Context, d *schema.ResourceData, meta interface{})
 		return diag.Errorf("failed to get API endpoints for LKE cluster %d: %s", id, err)
 	}
 
-	flattenedControlPlane := flattenLKEClusterControlPlane(cluster.ControlPlane)
+	acl, err := client.GetLKEClusterControlPlaneACL(ctx, id)
+	if err != nil {
+		if lerr, ok := err.(*linodego.Error); ok &&
+			lerr.Code == 400 && strings.Contains(lerr.Message, "Cluster does not support Control Plane ACL") {
+			// The cluster does not have a Gateway
+			tflog.Warn(ctx, fmt.Sprintf("This cluster does not have a Control Plane ACL. %v", err.Error()))
```

Comment:
Given this is in the read routine this seems to be a valid case, so we may not need to warn here 👍